store/sqlstore/thread_store.go

```diff
@@ -106,3 +114,69 @@ func (s *SqlThreadStore) Delete(threadId string) error {
 
 	return nil
 }
+
+func (s *SqlThreadStore) SaveMembership(membership *model.ThreadMembership) (*model.ThreadMembership, error) {
+	if err := s.GetMaster().Insert(membership); err != nil {
+		return nil, errors.Wrapf(err, "failed to save thread membership with postid=%s userid=%s", membership.PostId, membership.UserId)
+	}
+
+	return membership, nil
+}
+
+func (s *SqlThreadStore) UpdateMembership(membership *model.ThreadMembership) (*model.ThreadMembership, error) {
+	if _, err := s.GetMaster().Update(membership); err != nil {
+		return nil, errors.Wrapf(err, "failed to update thread membership with postid=%s userid=%s", membership.PostId, membership.UserId)
+	}
+
+	return membership, nil
+}
+
+func (s *SqlThreadStore) GetMembershipsForUser(userId string) ([]*model.ThreadMembership, error) {
+	var memberships []*model.ThreadMembership
+	_, err := s.GetReplica().Select(&memberships, "SELECT * from ThreadMemberships WHERE UserId = :UserId", map[string]interface{}{"UserId": userId})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get thread membership with userid=%s", userId)
+	}
+	return memberships, nil
+}
+
+func (s *SqlThreadStore) GetMembershipForUser(userId, postId string) (*model.ThreadMembership, error) {
+	var membership model.ThreadMembership
+	err := s.GetReplica().SelectOne(&membership, "SELECT * from ThreadMemberships WHERE UserId = :UserId AND PostId = :PostId", map[string]interface{}{"UserId": userId, "PostId": postId})
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, store.NewErrNotFound("Thread", postId)
+		}
+		return nil, errors.Wrapf(err, "failed to get thread membership with userid=%s postid=%s", userId, postId)
+	}
+	return &membership, nil
+}
+
+func (s *SqlThreadStore) DeleteMembershipForUser(userId string, postId string) error {
+	if _, err := s.GetMaster().Exec("DELETE FROM ThreadMemberships Where PostId = :PostId AND UserId = :UserId", map[string]interface{}{"PostId": postId, "UserId": userId}); err != nil {
+		return errors.Wrap(err, "failed to update thread membership")
+	}
+
+	return nil
+}
+
+func (s *SqlThreadStore) UpdateMembershipFromMention(userId, postId string) error {
+	membership, err := s.GetMembershipForUser(userId, postId)
+	if err != nil {
+		var nfErr *store.ErrNotFound
+
+		if !errors.As(err, &nfErr) {
+			return errors.Wrap(err, "failed to get thread membership")
+		}
+		_, err = s.SaveMembership(&model.ThreadMembership{
+			PostId:      postId,
+			UserId:      userId,
+			Following:   true,
+			LastViewed:  0,
+			LastUpdated: 0,
+		})
+		return err
+	}
+	_, err = s.UpdateMembership(membership)
```

Comment:
If there is not error getting the membership we are storing again the same membership, I think makes no sense unless you process it, actually. I think if we are leveraging LastUpdated to have Sync api endpoint, probably we want to update the membership if the following is false, to true, and the LastUpdated if that happens.